docs/proposals/images/machinepool-machines/inframachinepool-delete-replicas.plantuml

```diff
@@ -0,0 +1,33 @@
+@startuml inframachinepool-delete-replicas
+:InfraMachinePool delete replicas reconciliation;
+:fetch all InfraMachinePoolMachines;
+:fetch owner Machine for each InfraMachinePoolMachine;
+if (any owner Machine is missing) then
+  :wait for owner Machine to be created;
+  :requeue;
+  stop
+else
+endif
+repeat 
+  if (replica backing Machine does not exist) then (yes)
```

Comment:
The diagram makes it look like we go from reconciling all machines to looking at a specific Machine here, how do we know which Machine this is talking about?

Reply:
I tried to restructure this into a for-each loop. Let me know if this looks better!